common/src/types.rs

```diff
@@ -71,18 +71,24 @@ pub struct AgendaProof {
     pub proof: Vec<TypedSignature<Agenda>>,
 }
 
+/// An abstracted diff of the state.
+///
+/// - The actual content of the diff (for the non-reserved state)
+/// is not cared by the Simperby node. It only keeps the hash of it.
+/// - It holds the reserved state as a `Box` to flatten the variant size.
+/// (see https://rust-lang.github.io/rust-clippy/master/index.html#large_enum_variant)
 #[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
 pub enum Diff {
     /// Nothing changed in the repository; an empty commit.
     None,
-    /// Only changes non-reserved areas. Contains the hash of the diff.
+    /// Changes the reserved area ONLY.
+    Reserved(Box<ReservedState>),
+    /// Changes the non-reserved area ONLY.
     ///
-    /// The actual content of the diff is not covered by this crate; see `simperby-repository`.
-    General(Hash256),
-    /// Changes the reserved area. Contains the new reserved state and the hash of the diff.
-    /// It holds the reserved state as a `Box` to flatten the variant size.
-    /// (see https://rust-lang.github.io/rust-clippy/master/index.html#large_enum_variant)
-    Reserved(Box<ReservedState>, Hash256),
+    /// It contains the hash of the diff.
+    NonReserved(Hash256),
+    /// General diff that may change both the reserved state and the non-reserved state.
+    General(Box<ReservedState>, Hash256),
```

Comment:
Currently there is no tests for `Diff::General`. How about leaving a TODO comment?

Or maybe not necessary since it can be considered as union of `Diff::Reserved` and `Diff::NonReserved`?

Reply:
Yeah, there is no test that covers the `General` variant.
However, we don't really comment all code components that need to be tested, do we?
I think it's more suitable to be an issue.

Reply:
#158